agent/php_call.c

```diff
@@ -48,8 +52,29 @@ zval* nr_php_call_user_func(zval* object_ptr,
 
   fname = nr_php_zval_alloc();
   nr_php_zval_str(fname, function_name);
-#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO /* PHP 8.0+ */
+#if ZEND_MODULE_API_NO >= ZEND_8_2_X_API_NO /* PHP 8.2+ */
+  /*
+   * With PHP 8.2, functions that do not exist will cause a fatal error to
+   * be thrown. `zend_call_method_if_exists` will attempt to call a function and
+   * silently fail if it does not exist
+   */
+  if (NULL != object_ptr) {
+    object = Z_OBJ_P(object_ptr);
+  } else {
+    object = NULL;
```

Comment:
Since object is already assigned NULL [here](https://github.com/newrelic/newrelic-php-agent/pull/621/files#diff-92a37f49fed3190bcf77a05cd9b996d03bc1a904264d7131a86b64a4860b1ccaR19) is this else statement necessary?

Reply:
No, technically it isn't- I believe when we were going through this fix it was added for developer clarity, but isn't necessary.

Reply:
Fixed in d340b89bd3739c2c2b91ba7051ee1e2434f05ff5

Reply:
I think this change is dangerous - if someone sees there is an `object` variable to use above this code and assign it to something other than NULL then this existing code will not work properly.  I don't think it is good to assume someone will catch this side effect so being defensive and setting it NULL is just being safe.

Reply:
If someone sees `object` and assigns it to a non-null value, it will just be overwritten when it reaches this point, which is right before it gets passed to `zend_call_method_if_exists`, after which point it isn't used again. Given that I think it would take a degree of intentional effort to break this functionality.
That said I don't have an issue restoring the `else` statement to make the intention clear to the developer, even if it is strictly unnecessary from a code point of view.
@zsistla - are you comfortable with @mfulb's suggestion or do you prefer the `else` statement be removed?

Reply:
If someone assigns it to something other than NULL and we overwrite it with NULL (possibly not freeing it), wouldn't that also cause issues?  I'm okay with returning the else statement, it wasn't a blocker for me either way.

Reply:
Reverted in 6249464d66a8ef6a2acca661ac7187cf6f21dcf4.